gcp/appengine/blog/content/posts/using-the-determineversion-api.md

```diff
@@ -0,0 +1,184 @@
+---
+title: "Using the determineversion API to find C/C++ vulnerabilities"
+date: 2023-07-18T19:16:34Z
+draft: true
+author: OSV Team
+---
+With the increasing incidence of software supply chain attacks, it is more important than ever for developers to understand the known vulnerabilities in their open source dependencies, regardless of the ecosystem of origin. Within the C/C++ ecosystem it is difficult to match dependencies to vulnerabilities for a few of reasons:
+
+- C/C++ does not have a centralized package manager like npm or pyPI
+- Software projects typically pull in C/C++ by submodules or vendoring
+- Source code identifiers (e.g. git hashes) are the best way to identify libraries, but vulnerabilities are typically associated to versions, not git hashes
+
+OSV has had C/C++ vulnerability data from OSS-Fuzz keyed on git hashes from day 1. However, a remaining challenge for C/C++ users is being able to accurately identify the closest upstream git hash of their C/C++ dependencies in order to make use of this vulnerability data.The OSV team is committed to bridging the gap between what C/C++ users need and the constraints of the ecosystem and today we are announcing a tool that can help–the determineversion API. 
+<!--more-->
+
+## What is the determineversion API?
+The [determineversion API](https://google.github.io/osv.dev/post-v1-determineversion/) is a new, experimental API endpoint for OSV’s API. The goal of the determineversion API is to help users determine the likely version of their vendored C/C++ dependencies. Once dependency versions are known, our other API endpoints can return reliable vulnerability information. 
+
+## What projects can use the determineversion API?
+The set of C/C++ repositories currently supported by the determineversion API are the ones being fuzzed in OSS-Fuzz. This is because OSV currently only contains commit-level vulnerability information for C/C++ projects via OSS-Fuzz. If a project has been fuzzed by OSS-Fuzz, you’ll be able to use the tool, but you should understand that there may be additional vulnerabilities in your dependencies that weren’t found by or reported through OSS-Fuzz. You can check a project’s eligibility by following the steps in our [docs](https://google.github.io/osv.dev/post-v1-determineversion/#available-libraries). 
+
+The OSV team is also working on expanding C/C++ coverage by attaching commit-level details to the NVC/CVE database.Once released, we are hopeful that many of these gaps will no longer exist. Even small security steps can make a positive difference, and we hope that the current limitations will not prevent you from trying the new API endpoint. 
```

Comment:
I'd reword this more around that the NVD/CVE support will offer an extremely comprehensive solution to C/C++ vulnerability management. 

The current one is not a "small step" -- OSS-fuzz is already fuzzing most of the top C/C++ packages out there. So I'd remove the last sentence in this paragraph completely.

Reply:
This is a fair point. I have rephrased.